compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/FakeOverrideGenerator.kt

```diff
@@ -67,9 +67,22 @@ class FakeOverrideGenerator(
     }
 
     fun IrClass.getFakeOverrides(klass: FirClass<*>, realDeclarations: Collection<FirDeclaration>): List<IrDeclaration> {
+        val classLookupTag = klass.symbol.toLookupTag()
+        val realDeclarationNames = realDeclarations.mapNotNullTo(mutableSetOf()) {
+            val callableDeclaration = (it as? FirCallableMemberDeclaration<*>) ?: return@mapNotNullTo null
+            // We may need to create a new fake override; compute base symbols; or deal with name conflicts.
+            if (callableDeclaration.symbol.shouldHaveComputedBaseSymbolsForClass(classLookupTag) ||
+                callableDeclaration.allowsToHaveFakeOverrideIn(klass) ||
+                callableDeclaration.visibility == Visibilities.Private
```

Comment:
Is this the case where there is already a `private fun foo` in the current class and maybe a `fun foo` in some super type, and we might need to create a fake override depending on the function signatures? If so, where do we actually ensure there is no conflict?

Reply:
Hm, I thought, for `foo` from a super type, we add a fake override, and then signature mangler detects a signature clash there. And I thought I saw failures if such `private` member in the current class is discarded that way can cause signature clash in backend. To make sure, I removed that and run all the tests, including full pipeline, and didn't see such failures again, huh. Maybe part of my checker trial caused that. 🤷  Removed here, thanks!